app/services/hyrax/workflow/sipity_actions_generator.rb

```diff
@@ -38,11 +39,39 @@ def call
       private
 
         def generate_state_diagram!
+          # look for old states that need to be removed and validate the states can be removed.
+          removed_states = validate_states_to_be_removed
+
+          # remove any actions that will no longer be needed by the workflow
+          remove_unused_actions
+
           actions_configuration.each do |configuration|
             Array.wrap(configuration.fetch(:name)).each do |name|
               StateMachineGenerator.generate_from_schema(workflow: workflow, name: name, **configuration.except(:name))
             end
           end
+
+          # remove any states that are no longer needed by the workflow
+          #  Note I am doing this after the update so nothing is still linking to the
+          #       state to keep it from being deleted.
+          removed_states.each(&:delete)
+        end
+
+        def validate_states_to_be_removed
+          new_states = actions_configuration.map { |a| a[:transition_to] }
+          removed_states = workflow.workflow_states.reject { |state| new_states.include?(state.name) }
+          removed_states.each do |state|
```

Comment:
@mjgiarlo Yeah that might be better, but since we have no validation some is better than none.  At this point I'm going to unassign myself from this ticket as there seems to be a lot of work left.  I'm off the sprint this week and I will let the next person pick back up with this work.